### PR TITLE
TOOLS-2802: Make mongo-tools-common a subpackage of mongo-tools

### PR DESCRIPTION
**Pull request by edobranov**
Jira: https://jira.mongodb.org/browse/TOOLS-2802

Pretty much all the changes are substituting a hyphen for a slash in the import lines, or adding the mtc files into mongo-tools.

I'll highlight a couple other notable changes below.

### CODE REVIEW

File: THIRD-PARTY-NOTICES [hunk 1]
```diff
@@ -1435,24 +1435,6 @@ License notice for github.com/mongodb/mongo-go-driver
    See the License for the specific language governing permissions and
    limitations under the License.
 
-----------------------------------------------------------------------
-License notice for github.com/mongodb/mongo-tools-common
```
Comment (edobranov) [inline THIRD-PARTY-NOTICES:1439]: I figured we don't need this anymore since we don't import mtc now.

File: common.yml [hunk 2]
```diff
@@ -281,23 +281,23 @@ mongo_tools_variables:
       mongod_args: ""
       mongod_port: 33333
     ssl: &mongod_ssl_startup_args
-      mongod_args: "--sslMode requireSSL --sslCAFile testdata/ca-ia.pem --sslPEMKeyFile testdata/test-server.pem"
-      replsettest_ssl_config: "sslMode: \\\"requireSSL\\\",sslPEMKeyFile: \\\"testdata/test-server.pem\\\", sslCAFile: \\\"testdata/ca-ia.pem\\\", sslAllowInvalidHostnames: \\\"\\\""
+      mongod_args: "--sslMode requireSSL --sslCAFile common/db/testdata/ca-ia.pem --sslPEMKeyFile common/db/testdata/test-server.pem"
+      replsettest_ssl_config: "sslMode: \\\"requireSSL\\\",sslPEMKeyFile: \\\"common/db/testdata/test-server.pem\\\", sslCAFile: \\\"common/db/testdata/ca-ia.pem\\\", sslAllowInvalidHostnames: \\\"\\\""
```
Comment (edobranov) [inline common.yml:285]: I was on the fence about where all the PEM files should live, but I think it's fine to leave them as-is in the `common` subpackage. Some testing code in `common` references them with a relative path and I felt it wasn't worth it to make changes there.

File: go.mod [hunk 3]
```diff
@@ -8,20 +8,21 @@ require (
...
 	github.com/google/go-cmp v0.5.2
 	github.com/gopherjs/gopherjs v0.0.0-20190430165422-3e4dfb77656c // indirect
-	github.com/jessevdk/go-flags v1.4.0 // indirect
+	github.com/jessevdk/go-flags v1.4.0
 	github.com/jtolds/gls v4.20.0+incompatible // indirect
 	github.com/klauspost/compress v1.10.1 // indirect
 	github.com/mattn/go-colorable v0.1.7 // indirect
 	github.com/mattn/go-runewidth v0.0.4 // indirect
 	github.com/mgutz/ansi v0.0.0-20200706080929-d51e80ef957d // indirect
-	github.com/mongodb/mongo-tools-common v4.0.18+incompatible
```
Comment (edobranov) [inline go.mod:17]: Just indication that we don't rely on `mongo-tools-common` anywhere now.

File: common.yml (hunk 2 above)
Comment (rychipman) [reply common.yml:285]: That seems sensible to me! Does this mean that we can now delete the PEM files that had been moved into the `/testdata/` dir?

**Review by rychipman — CHANGES_REQUESTED** [review]

File: common.yml (hunk 2 above)
Comment (edobranov) [reply common.yml:285]: Yep, although git is interpreting the deletion as a rename/move to the `common` directory.

If you ctrl+F for this string in the "Files changed" tab, both PEM files should show up:
```
.pem → common/db/testdata/
```

Comment (rychipman) [reply common.yml:285]: Ah, yep; I see it now. Thanks!

**Review by rychipman — APPROVED** [review]
LGTM!

**Review by mattChiaravalloti — APPROVED** [review]

**Pull request merged**